MTPot.py

```diff
@@ -119,6 +188,18 @@ def handleException(self, exc_type, exc_param, exc_tb):
         honey_logger.debug(traceback.format_exc())
         return True
 
+    #Catch tracestack error
+    def inputcooker(self):
+        try:
+            TelnetHandler.inputcooker(self)
+        except socket.timeout as e:
+            #print 'socket-'+str(self.client_address[0])+':'+str(self.client_address[1]), e
+            custom_pool.remove_connection(str(self.client_address[0])+':'+str(self.client_address[1]))
+            honey_logger.debug("[%s:%d] session timed out", self.client_address[0], self.client_address[1])
+            self.finish()
+        except socket.error as e:
```

Comment:
Bah.. Oh well.. So let's at least check that the error code matches the file descriptor error code and raise otherwise.